### PR TITLE
Streamline test environments (using tox) and docs build

### PR DESCRIPTION
**Pull request by yannikschaelte**
* Use tox to streamline local and on-server testing environments
* Docs:
  * Refactor and simplify readthedocs build instructions
  * Add intersphinx and viewcode
* Replace run_flake8.sh by standard .flake8 config file and `tox -e flake8` to allow general plugins in a separate environment
* Replace setup.py by setup.cfg (just because it's nicer), add more meta-data.
* Add pre-commit hooks to enhance code standardization
* Update contribution guide to all changes

### CODE REVIEW

File: doc/authors.rst [hunk 1]
```diff
@@ -12,4 +12,4 @@ This package was mainly developed by:
 - Leonard Schmiester
 - Dantong Wang
 - Leonard Schmiester
-- Caro Loos 
+- Caro Loos
```
Comment (FFroehlich) [inline doc/authors.rst:15]:
```suggestion
- Carolin Loos
```

File: doc/contribute.rst [hunk 2]
```diff
@@ -1,77 +1,129 @@
...
 
-Contribute tests
-----------------
+If you contribute to the development of pyPESTO, install developer requirements
+via::
 
-Tests are located in the ``test`` folder. All files starting with ``test_``
-contain tests and are automatically run on Travis CI. To run them manually,
-type::
+    pip install -r requirements-dev.txt
```
Comment (FFroehlich) [inline doc/contribute.rst:26]: add a short sentence that this will install the tools described below?

File: doc/contribute.rst [hunk 3]
```diff
@@ -1,77 +1,129 @@
...
-We try to respect the `PEP8 <https://www.python.org/dev/peps/pep-0008>`_
-coding standards. We run `flake8 <https://flake8.pycqa.org>`_ as part of the
-tests. If flake8 complains, the tests won't pass. You can run it via::
+    tox [-e flake8,doc]
+
+with optional ``-e`` options specifying the environments to run, see
+``tox.ini`` for details.
 
-    ./run_flake8.sh
+GitHub Actions
```
Comment (FFroehlich) [inline doc/contribute.rst:57]: Does this still belong to Environment?

File: test/petab/test_amici_prediction.py [hunk 4]
```diff
@@ -83,7 +83,7 @@ def create_intial_assignment(sbml_model, spec_id):
                                  model_output_dir,
                                  verbose=False,
                                  observables=observables,
-                                 constantParameters=constantParameters)
+                                 constant_parameters=constant_parameters)
```
Comment (FFroehlich) [inline test/petab/test_amici_prediction.py:86]: 👍

File: test/petab/test_amici_prediction.py [hunk 5]
```diff
@@ -83,7 +83,7 @@ def create_intial_assignment(sbml_model, spec_id):
...
 
         # Importing the module and loading the model
         sys.path.insert(0, os.path.abspath(model_output_dir))
```
Comment (FFroehlich) [inline test/petab/test_amici_prediction.py:89]: use `amici.import_model_module` here?

**Review by FFroehlich — APPROVED** [review]

File: doc/contribute.rst (hunk 3 above)
Comment (yannikschaelte) [reply doc/contribute.rst:57]: makes more sense stand-alone I think, agreed

File: doc/authors.rst (hunk 1 above)
Comment (yannikschaelte) [reply doc/authors.rst:15]: restructured all names now

File: test/petab/test_amici_prediction.py (hunk 5 above)
Comment (yannikschaelte) [reply test/petab/test_amici_prediction.py:89]: that is done one line below

**Pull request merged**